File: Tellem/src/main/java/com/example/tellem/application/models/TaskList.java
package com.example.tellem.application.models;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class TaskList {
    public String name;
    public List<Task> tasks = new ArrayList<>();

    public TaskList(){
        name = "lol";
    }

    public TaskList(String nameTmp){
        name = nameTmp;
    }

    public void addTask(Task task){
        tasks.add(task);
    }
    public void addTask(String name,String desc, LocalDate date1, LocalDate date2){
        tasks.add(new Task(name, desc, date1, date2));
    }
    public void addTask(String name, LocalDate date1, LocalDate date2){
        tasks.add(new Task(name, date1, date2));
    }
    public void addTask(String name, String desc, LocalDate date1){
        tasks.add(new Task(name, desc, date1));
    }
    public void addTask(String name, LocalDate date1){
        tasks.add(new Task(name, date1));
    }

    public void removeTask(int id){
        tasks.set(id, null);
        tasks.remove(id);
    }
}

File: Tellem/src/main/kotlin/tellem/model/TaskList.kt
package tellem.model

import java.time.LocalDate

class TaskList {
    var name: String?
    var tasks: MutableList<Task?> = ArrayList()

    constructor() {
        name = "Task list"
    }

    constructor(nameTmp: String?) {
        name = nameTmp
    }

    fun addTask(task: Task?) {
        tasks.add(task)
    }

    fun addTask(name: String?, desc: String?, date1: LocalDate?, date2: LocalDate?) {
        tasks.add(Task(name, desc, date1, date2))
    }

    fun addTask(name: String?, date1: LocalDate?, date2: LocalDate?) {
        tasks.add(Task(name, date1, date2))
    }

    fun addTask(name: String?, desc: String?, date1: LocalDate?) {
        tasks.add(Task(name, desc, date1))
    }

    fun addTask(name: String?, date1: LocalDate?) {
        tasks.add(Task(name, date1))
    }

    fun removeTask(id: Int) {
        tasks[id] = null
        tasks.removeAt(id)
    }
}